v2/protocol/http/utility.go

```diff
@@ -24,3 +26,63 @@ func NewEventFromHTTPResponse(resp *nethttp.Response) (*event.Event, error) {
 	msg := NewMessageFromHttpResponse(resp)
 	return binding.ToEvent(context.Background(), msg)
 }
+
+// NewEventsFromHTTPRequest returns a batched set of Events from a http.Request
+func NewEventsFromHTTPRequest(req *nethttp.Request) ([]event.Event, error) {
+	msg := NewMessageFromHttpRequest(req)
+	return binding.ToEvents(context.Background(), msg, msg.BodyReader)
+}
+
+// NewEventsFromHTTPResponse returns a batched set of Events from a http.Response
+func NewEventsFromHTTPResponse(resp *nethttp.Response) ([]event.Event, error) {
+	msg := NewMessageFromHttpResponse(resp)
+	return binding.ToEvents(context.Background(), msg, msg.BodyReader)
+}
+
+// NewHTTPRequestFromEvent creates a http.Request object that can be used with any http.Client for a singular event.
+func NewHTTPRequestFromEvent(ctx context.Context, url string, event event.Event) (*nethttp.Request, error) {
+	if err := event.Validate(); err != nil {
+		return nil, err
+	}
+
+	req, err := nethttp.NewRequestWithContext(ctx, nethttp.MethodPost, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	if err := WriteRequest(ctx, (*binding.EventMessage)(&event), req); err != nil {
+		return nil, err
+	}
+
+	return req, nil
+}
+
+// NewHTTPRequestFromEvents creates a http.Request object that can be used with any http.Client for sending
+// a batched set of events.
+func NewHTTPRequestFromEvents(ctx context.Context, url string, events []event.Event) (*nethttp.Request, error) {
+	// Sending batch events is quite straightforward, as there is only JSON format, so a simple implementation.
+	for _, e := range events {
+		if err := e.Validate(); err != nil {
+			return nil, err
+		}
+	}
+	var buffer bytes.Buffer
+	err := json.NewEncoder(&buffer).Encode(events)
```

Comment:
Just for my own education, here we just encode the events directly, but above for a single event we use `WriteRequest` which seems to have quite a bit more logic. Is any of that logic needed for batch?

Reply:
The only way to send batch is in a JSON encoded format, so there is no need for much of that logic.

Reply:
🤔 What is there to transform? From my reading of the batch spec, it's JSON in, JSON out.

Reply:
LOL I dunno. I just noticed that for the single event it has that transform stuff and I didn't want to miss anything. @lionelvillard any idea what this transform stuff is for and why we need it for single events but not batch?

Reply:
I figure at least we don't need it for the utility functions, since none of the existing functions in `utility.go` don't pass in transformers.

The good thing is, if there was a need to add an optional list of transformers to the method signature isn't going to break.